hi-ml-histopathology/src/histopathology/utils/report_utils.py

```diff
@@ -15,8 +15,51 @@
                                 get_tags_from_hyperdrive_run)
 
 
+def run_has_val_and_test_outputs(run: Run) -> bool:
+    """Checks whether the given run has both validation and test outputs files.
+
+    :param parent_run: The run whose outputs to check.
+    :raises ValueError: If the run does not have the expected output file(s).
+    :return: `True` if the run has validation and test outputs, `False` if it is a legacy run with
+        only test outputs.
+    """
+    outputs_filename = "test_output.csv"
+    val_outputs_filename = "val/" + outputs_filename
+    test_outputs_filename = "test/" + outputs_filename
+
+    outputs_folder = "outputs/"
+    all_available_files: List[str] = run.get_file_names()
+    output_files = [filename[len(outputs_folder):] for filename in all_available_files
+                    if filename.startswith(outputs_folder)]
+    if val_outputs_filename in output_files and test_outputs_filename in output_files:
+        return True
+    elif outputs_filename in output_files:
+        return False
+    else:
+        raise ValueError(f"Run {run.display_name} ({run.id}) does not have the expected files: {output_files}")
+
+
+def crossval_runs_have_val_and_test_outputs(parent_run: Run) -> bool:
+    """Checks whether all child cross-validation runs have both validation and test outputs files.
+
+    :param parent_run: The parent Hyperdrive run.
+    :raises ValueError: If any of the child runs does not have the expected output files, or if
+        some of the child runs have both outputs and some have only test outputs.
+    :return: `True` if all children have validation and test outputs, `False` if all children are
+        legacy runs with only test outputs.
+    """
+    have_val_and_test_outputs = [run_has_val_and_test_outputs(child_run) for child_run in parent_run.get_children()]
+    if all(have_val_and_test_outputs):
+        return True
+    elif not any(have_val_and_test_outputs):
+        return False
+    else:
+        raise ValueError(f"Parent run {parent_run.display_name} ({parent_run.id}) has children with and without "
```

Comment:
Would the user know what is meant by 'with and without val and test outputs, or should we explicitly say which 2 files we are looking for here?

Reply:
If this error is going to be printed out, we should probably also say which child runs are incomplete.

Reply:
Clarified the message a bit more. I was also trying to be a bit cautious about over-committing on the filenames, as they may still be subject to change.

Reply:
@fepegar This failure mode of mixed child runs should never happen, as Hyperdrive is managed by Azure ML and children execute with exactly the same code snapshot. If the expected files are indeed missing from any of the child runs, a `ValueError` will be raised instead from `run_has_val_test_outputs()` with the specific details.

Reply:
I think it makes it explicit that either all children are of type A or all of type B, otherwise something has gone horribly wrong :)